Flakey CLI Test PushServiceIT.Test_Push_New_Site

Failed nightly 

https://github.com/dotCMS/core/actions/runs/10987225840/job/30502081346


```
2024-09-23T04:07:56.7673125Z [INFO] Tests run: 24, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 52.42 s -- in com.dotcms.cli.command.site.SiteCommandIT
2024-09-23T04:08:01.5174527Z [INFO] 
2024-09-23T04:08:01.5178173Z [INFO] Results:
2024-09-23T04:08:01.5178646Z [INFO] 
2024-09-23T04:08:01.5179036Z [ERROR] Errors: 
2024-09-23T04:08:01.5180363Z [ERROR]   PushServiceIT.Test_Push_New_Site » TraversalTask Error creating folder [/folder4 withSpace/]
2024-09-23T04:08:01.5181401Z [INFO] 
2024-09-23T04:08:01.5181941Z [ERROR] Tests run: 229, Failures: 0, Errors: 1, Skipped: 1
2024-09-23T04:08:01.5182634Z [INFO]
```

[4_Nightly Test _ CLI Tests.txt](https://github.com/user-attachments/files/17095461/4_Nightly.Test._.CLI.Tests.txt)


Previous error probably related
https://github.com/dotCMS/core/actions/runs/10969323714/job/30462020446

```
2024-09-21T04:03:05.8959454Z [INFO] Results:
2024-09-21T04:03:05.8960931Z [INFO] 
2024-09-21T04:03:05.8961635Z [ERROR] Errors: 
2024-09-21T04:03:05.8962544Z [ERROR]   PushServiceIT.Test_Push_New_Site » TraversalTask Error creating folder [/folder1/]
2024-09-21T04:03:05.8966831Z [INFO] 
2024-09-21T04:03:05.8967618Z [ERROR] Tests run: 229, Failures: 0, Errors: 1, Skipped: 1
```

--- a/dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/business/HostAPIImpl.java
+++ b/dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/business/HostAPIImpl.java
@@ -121,23 +121,16 @@ public class HostAPIImpl implements HostAPI, Flushable<Host> {
     @Override
     @CloseDBIfOpened
     public Host resolveHostName(String serverName, User user, boolean respectFrontendRoles) throws DotDataException, DotSecurityException {
-        Host host = hostCache.getHostByAlias(serverName);
         User systemUser = APILocator.systemUser();
-
-        if(host == null){
-
-            try {
-                final Optional<Host> optional = resolveHostNameWithoutDefault(serverName, systemUser, respectFrontendRoles);
-                host = optional.isPresent() ? optional.get() : findDefaultHost(systemUser, respectFrontendRoles);
-            } catch (Exception e) {
-                host = findDefaultHost(systemUser, respectFrontendRoles);
-            }
-
-            if(host != null){
-                hostCache.addHostAlias(serverName, host);
-            }
+        Host host;
+        try {
+            final Optional<Host> optional =
+                    resolveHostNameWithoutDefault(serverName, systemUser, respectFrontendRoles);
+            host = optional.isPresent() ? optional.get() : findDefaultHost(systemUser, respectFrontendRoles);
+        } catch (Exception e) {
+            Logger.debug(this, "Exception resolving host using default", e);
+            host = findDefaultHost(systemUser, respectFrontendRoles);
         }
-
         checkSitePermission(user, respectFrontendRoles, host);
         return host;
     }

--- a/dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/business/HostAPIImpl.java
+++ b/dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/business/HostAPIImpl.java
@@ -121,15 +121,11 @@ public class HostAPIImpl implements HostAPI, Flushable<Host> {
     @Override
     @CloseDBIfOpened
     public Host resolveHostName(String serverName, User user, boolean respectFrontendRoles) throws DotDataException, DotSecurityException {
-        Host host;
-        final Host cachedHostByAlias = hostCache.getHostByAlias(serverName);
-        if (UtilMethods.isSet(() -> cachedHostByAlias.getIdentifier())) {
-            if (HostCache.CACHE_404_HOST.equals(cachedHostByAlias.getIdentifier())) {
-                return null;
-            }
-            host = cachedHostByAlias;
-        } else {
-            final User systemUser = APILocator.systemUser();
+        Host host = hostCache.getHostByAlias(serverName);
+        User systemUser = APILocator.systemUser();
+
+        if(host == null){
+
             try {
                 final Optional<Host> optional = resolveHostNameWithoutDefault(serverName, systemUser, respectFrontendRoles);
                 host = optional.isPresent() ? optional.get() : findDefaultHost(systemUser, respectFrontendRoles);
@@ -137,15 +133,12 @@ public class HostAPIImpl implements HostAPI, Flushable<Host> {
                 host = findDefaultHost(systemUser, respectFrontendRoles);
             }
 
-            if (host != null) {
+            if(host != null){
                 hostCache.addHostAlias(serverName, host);
-            } else {
-                hostCache.addHostAlias(serverName, HostCache.cache404Contentlet);
             }
         }
-        if (host != null) {
-            checkSitePermission(user, respectFrontendRoles, host);
-        }
+
+        checkSitePermission(user, respectFrontendRoles, host);
         return host;
     }
 

--- a/tools/dotcms-cli/cli/src/test/java/com/dotcms/api/client/files/PushServiceIT.java
+++ b/tools/dotcms-cli/cli/src/test/java/com/dotcms/api/client/files/PushServiceIT.java
@@ -200,8 +200,7 @@ class PushServiceIT {
      *
      * @throws IOException if an I/O error occurs
      */
-    // Temporarily ignore this test until the root cause of the problem is found
-    //@Test
+    @Test
     void Test_Push_New_Site() throws IOException {
 
         // Create a temporal folder for the pull

